Sources/KlaviyoSwift/StateManagement.swift

```diff
@@ -0,0 +1,293 @@
+                return .none
+            }
+            // TODO: check if we already have this token, skip sending if we do.
+            state.pushToken = pushToken
+            return .task { [state] in
+                return .enqueueRequest(try state.buildTokenRequest())
+            }
+        case .enqueueRequest(let request):
+            guard state.initialized else {
+                return .none
+            }
+            state.queue.append(request)
+            return .none
+        case .flushQueue:
+            guard state.initialized else {
+                return .none
+            }
+            if state.flushing {
+                return .none
+            }
+            if state.queue.isEmpty {
+                return .none
+            }
+            state.requestsInFlight.append(contentsOf: state.queue)
+            state.queue.removeAll()
+            state.flushing = true
+            return .task {
+                .sendRequest
+            }
+        case .stop:
+            guard state.initialized else {
+                return .none
+            }
+            return EffectPublisher.cancel(ids: [RequestId.self, FlushTimer.self])
+                .concatenate(with: .run(operation: { send in
+                    await send(.cancelInFlightRequests)
+                    await send(.archiveCurrentState)
+                }))
+        case .start:
+            guard state.initialized else {
+                return .none
+            }
+            return environment.analytics.timer(state.flushInterval)
+                    .map { _ in
+                        KlaviyoAction.flushQueue
+                    }.eraseToEffect()
+                    .cancellable(id: Timer.self, cancelInFlight: true)
+        case .dequeCompletedResults(let completedRequest):
+            state.requestsInFlight.removeAll { inflightRequest in
+                completedRequest.uuid == inflightRequest.uuid
+            }
+            if state.requestsInFlight.isEmpty {
+                state.flushing = false
+                return .none
+            }
+            return .task { .sendRequest }
+        case .sendRequest:
+            guard state.initialized else {
+                return .none
+            }
+            guard state.flushing else {
+                return .none
+            }
+            guard let request = state.requestsInFlight.first else {
+                state.flushing = false
+                return .none
+            }
+            return .run { send in
+                let result = await environment.analytics.klaviyoAPI.send(request)
+                switch result {
+                case .success(_):
+                    // TODO: may want to inspect response further.
+                    await send(.dequeCompletedResults(request))
+                case .failure(_):
+                    // TODO: depending on failure may want to deque
+                    await send(KlaviyoAction.cancelInFlightRequests)
```

Comment:
I probably should limit retries here...

Reply:
Right now, if someone's network drops and the request fails is it just gonna attempt resending indefinitely?